### DISCUSSION TITLE
Pulsar Portable on Windows: Always opens Project, Welcome, Welcome Guide and Change Log (how to prevent this?)

### DISCUSSION THREAD

**Discussion by bernd-wechner** [pulsar-edit/.github#202] [Pulsar Support] [answered]
Just moving from Atom to Pulsar on my windows box and opted for portable (always do, as I it's in office and installed software is audited).

Up front, warm thanks for keep Atom, er Pulsar, alive. Truly the best portable, cross platform editor I've ever known. 

Alas after a few settings tweaks:

File > Settings > Core > Restore Previous Windows on Start = Always
File > Settings > Core > Open Empty Editor of Start = unchecked

Ever time I open Pulsar it crowds my whole window out with the Project, Welcome, Welcome Guide and Change Log panes (4 of them).

It does open also with one small pane that has the file in it that I had open when I closed Pulsar. So the state save is working. But it's not saving the state of these closed startup panes. I'd like them to stay closed. They do on my Pulsar install on Linux, and on Atom in both contexts.  I wonder if there's a setting I've missed or something awry?

**Comment by confused-Techie** [comment]
It sounds like some state is saved, while the rest is not.

Wanting to make sure, have you created a `.pulsar` directory right next to the directory that has the portable exe? The [docs](https://pulsar-edit.dev/docs/launch-manual/sections/getting-started/#portable-mode) detail this further, but this is where all package configuration data should be saved, such as when you change the settings like:

* `core: Restore Previous Windows on Start`: Always
* `core: Open Empty Editor on Start`: false
* `welcome: Show On Startup`: false
* `welcome: Show Change Log`: false

**Reply by bernd-wechner** [reply]
No, I had not.  Did not catch that. Would have imagined (and suggested in case the team wants that) that such a folder be created automatically on first run if possible.

But testing your suggestions (for which I am inordinately grateful, thank you):

1. I created the .pulser directory, and sure enough if I run Pulsar, change settings and open it again, that directory now has stuff in it. 
2. It does appear to remember the settings and also to open the file I had open, but continues to open the 4 undesired panes
3. I can't see a "welcome" section on the Settings pane at all (where are these?) 
4. On the settings pane there is a button labelled "Open config folder" which, when clicked, does nothing. Fails silently to do anything.
5. File > config does open a confid file and I added this as a quick guess:
  welcome:
    ShowOnStartup: "false"
    ShowChangeLog: "false"
but no change. Same behaviours.

**Reply by confused-Techie** [reply, marked as answer]
Glad to hear we are getting closer to desired behavior.

It's good to keep in mind every package can have it's own settings, and with how much of Atom and thus Pulsar relies on packages for it's base behavior, many of our settings are hidden away in those packages.

If you go to "Settings > Packages > welcome >" then you can find all the settings relevant to the `welcome` package. Which in this case controls those other features. (While these could absolutely be handled by manually adding them to the config file, I generally wouldn't suggest it, as the structure of the file is rather abstracted).

But if you don't want to dig through all packages to find your setting, that's why we now have the "Search" pane in settings, you can search for any given setting, and it'll do it's best to take you there. Where you can search, for example, `showChangeLog` and the first result should be from the `welcome` package:

![image](https://github.com/pulsar-edit/.github/assets/26921489/f377f2a1-f4f3-4450-8eb4-1bd07ede714f)

From there, clicking it will take you to the `welcome` package's settings page. Where you can change the settings I mentioned above, prefixed with `welcome:`

**Reply by bernd-wechner** [reply]
Awesome, thanks so much, that almost fixed everything. And my guesses weren't far off on the configs ;-). 

If I could only stop it showing the Project pane on startup now. I never could work out how to stop that in Atom, and maybe I'm stuck with it ;-). I did search "project" in the settings.  Personally, it's just in the way of using Atom/Pulsar as a great text editor. I expect that pane to open if I've opened a folder, but not if I open a file for example.

**Reply by confused-Techie** [reply]
If I'm understanding what you mean by the project pane, you may be looking for:

* `tree-view: Hidden On Startup`: false

But if that doesn't do it for you, I'm sure there's a community package that could help out!

Glad to see this nearly covers everything though!

**Reply by bernd-wechner** [reply]
You rock! That did it. That's an old Atom bugbear of mine. Not only is it fixed, I have learned a great deal in the process (about the package based nature of the editor). But yes, the tree-view is labelled at top "Projects". Hence the confusion (and lack of success in divining the name). I did see tree-view but it didn't click as that. I somehow had internalised an image of that as a widget inside the projects pane (call that a perception bias from other projects).

**Reply by confused-Techie** [reply]
Fantastic to hear! But no worries, not everything is as obvious as it should, part of us doing such an overhaul on the docs. 

But glad to hear things are looking a lot more like you prefer! Feel free to always get back in contact if there's a concern, happy coding

**Reply by bernd-wechner** [reply]
And ironically, you call yourself the confused techie ;-). He, he ... (it was me that was confused). But I was so sad when MS bought and then shafted Atom, it really was the best (and now Pulsar is!). So eminently configurable and portable. In my office everyone swears by Notepad++, and it is pretty good I'll grant, but it's a Windows app and I think that (and by inference it) has seen its day. In the 2020s now it is so easy to write cross-platform apps that I see few excuses, bar laziness, not to anymore. Mac and Linux users across the world rejoice at this trend (and have no doubt been instrumental in its evolution).  The beauty of cross-platform is many of us who work cross-platform have a consistency of tools of course (I mean VS Code is cross-platform, but I see it as an IDE, and while Atom/Pulsar did pitch that way a bit, it remains just a great configurable, powerful, portable text editor to many of us too and above all, it is community driven and poses no vendor lock-in threats - say all those folk mildly nervous about pinning their futures on VS-code - though the boundaries are definitely blurring given MS open sourced VS-code and kick-started at least with a very liberal license).

**Reply by confused-Techie** [reply]
Trust me, I'm confused more often than not. But thanks for the kind words, that's why we are all here. Thought it was to great of a thing to stop existing. But while I'm not the most eloquently spoken, very much agree that inclusivity of all platforms is the way forward, aiming for feature parity in all senses
